lib/deploy.js

```diff
@@ -18,47 +19,67 @@ function sftpDeploy(opts) {
     // Log that it was successful
     tessel.logs.info('Connected.');
 
-    tessel.logs.info('Bundling up code...');
-    // Command V2 to extract a tarball it receives on stdin to the remote deploy dir
-    conn.exec('mkdir -p /tmp/remote-script/; rm -rf /tmp/remote-script/*; tar -x -C /tmp/remote-script', function(err, rstdin) {
-
-      rstdin.on('error', function(e) {
-        tessel.logs.error("Unable to deploy code:", e);
-        process.exit(1);
-      });
+    // Stop any running processes and remove old files
+    conn.exec('/etc/init.d/tessel-app stop; rm -rf ' + filepath, function(err, stdin) {
+      stdin.on('close', function () {
+        tessel.logs.info('Bundling up code...');
+        // Command V2 to extract a tarball it receives on stdin to the remote deploy dir
+        conn.exec('mkdir ' + filepath + '/; tar -x -C ' + filepath, function(err, rstdin) {
+          rstdin.on('error', function(e) {
+            tessel.logs.error("Unable to deploy code:", e);
+            process.exit(1);
+          });
 
-      // Gather details about the file structure of the script being sent
-      var ret = tessel.analyzeScript(process.cwd() + "/" + opts.entryPoint, {verbose: opts.verbose});
-      // Tar up the code to improve transfer rates
-      tessel.tarCode(ret.pushdir, {node: true}, function(err, bundle) {
-        tessel.logs.info('Bundled.')
-        // Throw any unfortunate errors
-        if (err) throw err;
+          // Gather details about the file structure of the script being sent
```

Comment:
Small nitpick: Can you remove these tabs so these lines aren't registered as being changed?